pulsar-sql/presto-pulsar/src/main/java/org/apache/pulsar/sql/presto/PulsarRecordCursor.java

```diff
@@ -235,113 +237,117 @@ public void setPulsarSqlSchemaInfoProvider(PulsarSqlSchemaInfoProvider schemaInf
     }
 
     @VisibleForTesting
-    class DeserializeEntries implements Runnable {
+    class DeserializeEntries extends Thread {
 
         protected boolean isRunning = false;
 
-        private final Thread thread;
+        private final CompletableFuture<Void> closeHandle;
 
         public DeserializeEntries() {
-            this.thread = new Thread(this, "derserialize-thread-split-" + pulsarSplit.getSplitId());
+            super("deserialize-thread-split-" + pulsarSplit.getSplitId());
+            this.closeHandle = new CompletableFuture<>();
         }
 
-        public void interrupt() {
+        public CompletableFuture<Void> close() {
             isRunning = false;
-            thread.interrupt();
-        }
-
-        public void start() {
-            this.thread.start();
+            super.interrupt();
```

Comment:
Do we need to avoid the `DeserializeEntries` can being closed multiple times? Otherwise, I think we will get an exception here.

Reply:
Yes, right. fixed.